### DISCUSSION TITLE
Data Import Tool - CSV import not working

### DISCUSSION THREAD

**Discussion by frestalink** [firefly-iii/firefly-iii#5385] [Firefly III]
I'm new user here, so I'm still trying understand many things. I'm not sure what is missing to Data Import work...

in the LOG the message is 'isReadyForBasicStep("service-validation")', don't must be 'isReadyForCSVStep("service-validation")'
[2021-12-11 13:53:35] local.DEBUG: isReadyForBasicStep("service-validation")  

[env.txt](https://github.com/firefly-iii/firefly-iii/files/7697444/env.txt)
[data-import-2021-12-11.log](https://github.com/firefly-iii/firefly-iii/files/7697445/data-import-2021-12-11.log)

 Using:
Version 5.6.5  - https://docs.firefly-iii.org/firefly-iii/installation/self_hosted/
Firefly III Data Import Tool, v0.2.0 - https://docs.firefly-iii.org/data-importer/install/self_hosted/

**Comment by JC5** [comment]
Do me a favor and press the Flush button a few times, that could work.

**Comment by JC5** [comment]
This is not an error btw so I'm not sure what's not working exactly?

**Reply by frestalink** [reply]
When I press 'Import CSV file', nothing happens.
The button turns a 'gear' and goes back to normal, that's all.

**Reply by JC5** [reply]
Mmm. I just released 0.3.0 today, could you see if that one works better?

**Reply by frestalink** [reply]
Updated to 0.3.0, but the same issue.
The log is different, new messages.
[data-import-2021-12-11.log](https://github.com/firefly-iii/firefly-iii/files/7697639/data-import-2021-12-11.log)

**Reply by gillesbonnet** [reply]
> When I press 'Import CSV file', nothing happens. The button turns a 'gear' and goes back to normal, that's all.

I do have the same problem. I have installed today firefly with Ampps ([with great help from Ampps support](https://github.com/firefly-iii/firefly-iii/discussions/5199#discussioncomment-1852273)) and installed the importer (also with help from Ampps support). When I click on 'Import CSV files' it apparently reacts as if I click on 'start over': All three buttons turns a gear and goes back to normal. Note that the same happens with the other two buttons 'Import from Nordigen' and 'Import from Spectre'. I assume this is not the normal behavior. 
@JC5 Any idea what is going on, and what I should try to do?

**Reply by gillesbonnet** [reply]
I noted that I was using version 0.2.0. I have just upgraded it (successfully) to 0.3.0. Unfortunately it did not solve this issue.

**Reply by JC5** [reply]
I would like to see some logs so I can debug further.

**Reply by gillesbonnet** [reply]
> I would like to see some logs so I can debug further.

[data-import-2021-12-21.log](https://github.com/firefly-iii/firefly-iii/files/7756099/data-import-2021-12-21.log)
Does that help?

**Reply by JC5** [reply]
Unfortunately not. The "reauthenticate" button also doesn't help? In the mean time, I will add some debug logging.

**Reply by JC5** [reply]
Could you pull from the `develop` branch and try again?

**Reply by gillesbonnet** [reply]
> The "reauthenticate" button also doesn't help? 

no

**Reply by gillesbonnet** [reply]
> Could you pull from the `develop` branch and try again?

I'll try a bit later, but I'm not sure how to do that exactly.

**Reply by JC5** [reply]
No worries but you may have to wait for 0.4.0, which I hope to release before Christmas.

**Reply by gillesbonnet** [reply]
I upgraded to 0.4.0.  I have the same issue as before. Here is the log: [data-import-2021-12-22.log](https://github.com/firefly-iii/firefly-iii/files/7761164/data-import-2021-12-22.log)

**Reply by JC5** [reply]
You have an access token but no Firefly III url set?

**Reply by gillesbonnet** [reply]
one change compare to my experience with 0.3.0. The reauthenticate button is working now.

**Reply by gillesbonnet** [reply]
> You have an access token but no Firefly III url set?

I do have both, I believe.

**Reply by gillesbonnet** [reply]
ah wait it depends by what you refer to for the Firefly III url. I have filled the field Firefly III url set of http://127.0.0.1/data-importer-0-4-0/public/token with http://127.0.0.1/firefly/public/, but in the .env file the line 32 consists only of FIREFLY_III_URL= without any value. Should I add something here?

**Reply by gillesbonnet** [reply]
Now I have set FIREFLY_III_URL=http://127.0.0.1/firefly/public and FIREFLY_III_CLIENT_ID=2 in the .env file. The effect is that when I click on Reauthenticate it behaves (apparently) again as Start over.

Just in case you need it. I add the latest log: [data-import-2021-12-22.log](https://github.com/firefly-iii/firefly-iii/files/7761367/data-import-2021-12-22.log)

**Reply by JC5** [reply]
I see what you mean.  Does Firefly III work correctly in the sub directory you have now? That could be the issue.

**Reply by gillesbonnet** [reply]
Yes Firefly III works correctly.

**Reply by JC5** [reply]
I'll put something in the `develop` branch to validate the data, but I'm running out of ideas.

I can however recommend you use local subdomains instead of directories.

**Comment by JC5** [comment]
I see no error in the log files so there isn't much I can do for you. But are you sure this URL is correct? `http://firefly.localhost/public`

I suspect you need to configure your local environment to use `/` as a base.

**Reply by frestalink** [reply]
Just to close this, is solved.
I'm still learning, for this porpuse I decided install on **Windows 10** whit **XAMPP**.
I'm happy with the Firefly III so far.

So, for me, what I had to do after install *Firefly III* and *Data-Importer*:
#### 1) Configure Virtual Host
- Edit `apache/conf/httpd.conf` and enable this 2 lines
*Include conf/extra/httpd-vhosts.conf*
*LoadModule vhost_alias_module modules/mod_vhost_alias.so*

- Edit `apache/conf/extra/httpd-vhosts.conf`
I chose the ServerName `finance.firefly.com` and `importer.firefly.com`

    `<VirtualHost *:80>`
        `ServerAdmin webmaster@finance.firefly.com`
        `DocumentRoot "C:/xampp/htdocs/firefly-iii/public"`
        `ServerName finance.firefly.com`
        `ErrorLog "logs/finance.firefly.com-error.log"`
        `CustomLog "logs/finance.firefly.com-access.log" common`
    `</VirtualHost>`
    `<VirtualHost *:80>`
        `ServerAdmin webmaster@importer.firefly.com`
        `DocumentRoot "C:/xampp/htdocs/data-importer/public"`
        `ServerName importer.firefly.com`
        `ErrorLog "logs/importer.firefly.com-error.log"`
        `CustomLog "logs/importer.firefly.com-access.log" common`
    `</VirtualHost>`

- Edit `C:\Windows\System32\drivers\etc\hosts` and add
127.0.0.1       finance.firefly.com
127.0.0.1       importer.firefly.com

#### 2) Configure Data Importer
Edit `data-importer/.env` and set
- FIREFLY_III_URL=`http://127.0.0.1/firefly-iii/public`
- VANITY_URL=`http://finance.firefly.com`
- Access Firefly III on `http://finance.firefly.com`, see **Create new token** on documentation - https://docs.firefly-iii.org/data-importer/install/configure/
FIREFLY_III_ACCESS_TOKEN=`your_token`
- Still on Firefly III you must **Create new client**
Redirect URL = `http://importer.firefly.com/callback`

Good to go *Firefly III  Data Importer* on `http://importer.firefly.com`

Hope this helps someone!
